packages/block-library/src/post-excerpt/edit.js

```diff
@@ -43,11 +43,39 @@ export default function PostExcerptEditor( {
 } ) {
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
 	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
-	const [
-		rawExcerpt,
-		setExcerpt,
-		{ rendered: renderedExcerpt, protected: isProtected } = {},
-	] = useEntityProp( 'postType', postType, 'excerpt', postId );
+
+	const { editEntityRecord } = useDispatch( coreStore );
+	const setExcerpt = useCallback(
+		( newValue ) => {
+			void editEntityRecord( 'postType', postType, postId, {
+				excerpt: newValue,
+			} );
+		},
+		[ editEntityRecord, postType, postId ]
+	);
+	const { rawExcerpt, renderedExcerpt, isProtected } = useSelect(
+		( select ) => {
+			const record = select( coreStore ).getEntityRecord(
+				'postType',
+				postType,
+				postId,
+				{ excerpt_length: excerptLength }
+			);
+			const editedRecord = select( coreStore ).getEditedEntityRecord(
+				'postType',
+				postType,
+				postId
+			);
```

Comment:
I still need to test my theory here, but have you checked this change with Query blocks? I think because of the new argument; selectors won't be able to use data from `getEntityRecrods`, which will trigger separate HTTP requests.

Not a big issue in separation, but it can degrade performance inside the loop.

Reply:
No I haven't tested this with the Query block. But a few extra HTTP requests in the editor seem unavoidable this way (frontend is not affected). And I am not here to fundamentally change how selectors or the Query block works, that would be way out of scope for this PR. But if there's any quick wins or so, I'm eager to learn more.

Reply:
> And I am not here to fundamentally change how selectors or the Query block works, that would be way out of scope for this PR.

I wasn't suggesting that. I know it's not in this scope.

I'll try experimenting with it, but I can't promise anything until early next week.

P.S. We should simplify the data selector "orchestration" for consumers. Currently, improving performance for those requires familiarity with internals. It's okay for core but could be challenging for 3rd party consumers.

Reply:
Confirmed. Adding a new `excerpt_length` parameter triggers an additional request for each Excerpt block, which is not ideal, since it will increase editor loading time, but I guess it can't be avoided entirely.

Maybe we could mitigate this a bit by only applying the `excerpt_length` filter when `excerptLength !== DEFAULT_EXCERPT_LENGTH`.